### ISSUE TITLE
Add support for download attribute to link macro

### ISSUE THREAD

**Issue by jnorthr** [asciidoctor/asciidoctor#1781]
To suggest to a user's browser that a file download link is a stream file like .zip or .jar, we can compose html anchors using the 'download' keyboard. For example:

```
<a href="https://github.com/GPars/Website/blob/master/src/docs/resources/gpars-samples-1.2.1.zip" download>Zip file</a>
```

Browser then copies file into browser download folder rather than trying to display it.

Can we add an option to the asciidoctor link to declare these files? Something like:

```
https://github.com/GPars/Website/blob/master/src/docs/resources/gpars-samples-1.2.1.zip[Zip file,opts=download]
```

**Comment by mojavelinux** [comment]
Related discussion thread: http://discuss.asciidoctor.org/How-to-Generate-DOWNLOAD-keyword-for-HTML-Anchor-td4711.html

See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-download for details about the download attribute on the `<a>` element.

**Comment by jnorthr** [comment]
Most excellent Dan, thank YOU ! Looking forward to 1.6.0 - how soon ?

**Comment by sanmai-NL** [comment]
See https://github.com/asciidoctor/asciidoctor/issues/1883#issuecomment-250965157

**Comment by fritz-hh** [comment]
I have tested the behaviour of chrome 69 and firefox 63 with the `download` attribute set using a PDF file as destination of the href:

* In case the html content is opened with chrome and firefox from a local filesystem: The `download` attribute has no effect (the browser tries to display the file inside the browser)
* In case the html content is server by a webserver, both firefox and chrome behave as expected when the attribute is set: The file is downloaded and the browser does not try to display it.

**Comment by TheConen** [comment]
Is this implemented yet? I need this behavior and have found no way to achieve it...

**Comment by mojavelinux** [comment]
If the issue is open, then it's not implemented. If you need help implementing it or crafting a workaround, please ask in the project chat at https://chat.asciidoctor.org.

**Comment by mojavelinux** [comment]
I've discovered that if the value of the `href` attribute is transferred to the download attribute and the `href` attribute is set to empty, it will really encourage the browser to present a download dialog. Given that there are different ways to do this, I think a `download` attribute in AsciiDoc makes sense. Something like:

* download= (or perhaps download=true) (will add the valueless `download` attribute)
* download=only (or perhaps download=force) (will set `download=<value of href>` and clear href)
* download=<custom path> will use the value as specified

In other words, we may need more than just an option here since there are different ways of markup up a link or xref as downloadable.

Keep in mind that this only works for same-origin URLs and not all browsers honor it. See https://caniuse.com/download
